packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -114,6 +158,19 @@ if (x) {
         <T extends boolean | null | undefined>(x: T) => (x ? 1 : 0);
       `,
     },
+    {
+      options: [{ allowNullableBoolean: true }],
+      code: `
+        declare const x: (boolean | null)[];
+        x.filter(function (t) {
+          return t;
+        });
+        x.filter(t => {
+          return t ? 1 : 0;
+        });
+        x.filter(t => !t);
```

Comment:
[Testing] We generally try to stick to one thing under test for each test case. For this one, it looks like two things are being tested:

* That the `allowNullableBoolean` option still works...
* ...and that general logic holds true for all three variations of something passed to `.filter`

I don't think the latter is necessary. We can trim the tests down a bit:

```suggestion
```

Here and the other ones with several variations of the same, such as `allowNullableString: true`.